central/group/datastore/datastore_test.go

```diff
@@ -64,44 +67,97 @@ func (s *groupDataStoreTestSuite) TearDownTest() {
 	s.mockCtrl.Finish()
 }
 
+func walkMockFunc(groups []*storage.Group) func(_ context.Context, fn func(group *storage.Group) error) error {
+	return func(_ context.Context, fn func(group *storage.Group) error) error {
+		for _, g := range groups {
+			if err := fn(g); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+}
+
 func (s *groupDataStoreTestSuite) TestEnforcesGet() {
-	s.storage.EXPECT().Get(gomock.Any()).Times(0)
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Times(0)
 
 	group, err := s.dataStore.Get(s.hasNoneCtx, &storage.GroupProperties{})
 	s.NoError(err, "expected no error, should return nil without access")
 	s.Nil(group, "expected return value to be nil")
 }
 
 func (s *groupDataStoreTestSuite) TestAllowsGet() {
-	s.storage.EXPECT().Get(gomock.Any()).Return(nil, nil)
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Return(nil, false, nil)
 
-	_, err := s.dataStore.Get(s.hasReadCtx, &storage.GroupProperties{})
+	_, err := s.dataStore.Get(s.hasReadCtx, &storage.GroupProperties{Id: "1"})
 	s.NoError(err, "expected no error trying to read with permissions")
 
-	s.storage.EXPECT().Get(gomock.Any()).Return(nil, nil).Times(2)
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Return(nil, false, nil).Times(2)
 
-	_, err = s.dataStore.Get(s.hasWriteCtx, &storage.GroupProperties{})
+	_, err = s.dataStore.Get(s.hasWriteCtx, &storage.GroupProperties{Id: "1"})
 	s.NoError(err, "expected no error trying to read with permissions")
 
-	_, err = s.dataStore.Get(s.hasWriteAccessCtx, &storage.GroupProperties{})
+	_, err = s.dataStore.Get(s.hasWriteAccessCtx, &storage.GroupProperties{Id: "1"})
 	s.NoError(err, "expected no error trying to read with Access permissions")
 }
 
+func (s *groupDataStoreTestSuite) TestGet() {
+	group := fixtures.GetGroup()
+	s.storage.EXPECT().Get(gomock.Any(), group.GetProps().GetId()).Return(group, true, nil)
+
+	// Test that can fetch by id
+	g, err := s.dataStore.Get(s.hasReadCtx, &storage.GroupProperties{Id: group.GetProps().GetId()})
+	s.NoError(err)
+	s.True(reflect.DeepEqual(g, group))
+}
+
+// TODO(ROX-11592): This can be removed once retrieving the group by its properties is fully deprecated.
+func (s *groupDataStoreTestSuite) TestGetFetchesByProps() {
+	groups := fixtures.GetGroups()
+	expectedGroup := groups[6]
+	expectedProps := expectedGroup.GetProps()
+	expectedProps.Id = "" // clear out the id for the purposes of this test
+	s.storage.EXPECT().Walk(gomock.Any(), gomock.Any()).DoAndReturn(walkMockFunc(groups))
+
+	// Test that can fetch by props
+	g, err := s.dataStore.Get(s.hasReadCtx, expectedGroup.GetProps())
+	s.NoError(err)
+	s.True(reflect.DeepEqual(g, expectedGroup))
```

Comment:
nit: should be able to just use `s.Equal(group,g)` here instead of relying on `reflect.DeepEqual`.